tests/integration/test_headers.py

```diff
@@ -0,0 +1,88 @@
+import pytest
+import requests
+
+
+HEADER_AUTHORIZATION = 'Authorization'
+HEADER_ECHO = 'echo' # enable echo target
+HEADER_BASE_URL = 'x-ers-network-baseurl'
+HEADER_USER_ID = 'x-ers-user-id'
+HEADER_X_REQUEST_ID = 'x-request-id'
+HEADER_ASID = 'xapi_asid'
+
+CORRELATION_ID = "CORRELATION_ID"
+BUSINESS_FUNCTION = "BUSINESS_FUNCTION"
+ODS_CODE = "ODS_CODE"
+FILENAME = "FILENAME"
+COMM_RULE_ORG = "COMM_RULE_ORG"
+REFERRAL_ID = "REFERRAL_ID"
+
+EXPECTED_REFERRAL_ID = '000000040032'
+EXPECTED_CORRELATION_ID = '123123-123123-123123-123123'
+EXPECTED_BUSINESS_FUNCTION = 'REFERRING_CLINICIAN'
+EXPECTED_ODS_CODE = 'R69'
+EXPECTED_FILENAME = 'mysuperfilename.txt'
+EXPECTED_ASID = '280477200122'
+EXPECTED_COMM_RULE_ORG = 'R100'
+EXPECTED_USERID = '0987654321123'
+
+ORIGINAL_HEADERS = {
+    CORRELATION_ID : "x-correlation-id",
+    BUSINESS_FUNCTION: "nhsd-ers-business-function",
+    ODS_CODE : "nhsd-ers-ods-code",
+    FILENAME : "nhsd-ers-file-name",
+    COMM_RULE_ORG : "nhsd-ers-comm-rule-org",
+    REFERRAL_ID : "nhsd-ers-referral-id"
+}
+
+RENAMED_HEADERS = {
+    CORRELATION_ID : "nhsd-correlation-id",
+    BUSINESS_FUNCTION : "x-ers-business-function",
+    ODS_CODE : "x-ers-ods-code",
+    FILENAME : "x-ers-xapi-meta-file_name",
+    COMM_RULE_ORG : "x-ers-xapi-comm-rule-org",
+    REFERRAL_ID : "x-ers-xapi-meta-intended_ubrn"
+}
+
+@pytest.mark.integration_test
+class TestHeaders:
+
+    @pytest.mark.parametrize('user_id,asid', [(EXPECTED_USERID,EXPECTED_ASID)])
```

Comment:
Parametrize is usually used to run your test over a list of values. I was meaning more refactor your CONSTs into here and run the test for each header but I think that's up to you really. If not and if it's just for the default userid and asid I'd take off the parametrize and bring in the userid and asid as it took me a while to realise where they were coming from.

Reply:
The user id and asid are passed in by the test such that the access coe and app can be configured per test as per the [pytest example here](https://docs.pytest.org/en/6.2.x/fixture.html#override-a-fixture-with-direct-test-parametrization). 
I can understand there's confusion where there's magic involved but once you understand the magic it can be really useful in the future. I am even thinking that if further values are needed like other apps, access codes, or products, then I could define a fixture inside the class that can then be used by the test and defines it's own behaviour in the same way by passing different values in.

I am actually quite happy with using such magic as being a Java developer you get to see magic happen often. The trick is to understand the magic and then be dazzled :)

Reply:
@acarriedev could you please close this requested change if you are happy with me to keep it as-is?